docs/framework/windows-workflow-foundation/debugging-workflows.md

```diff
@@ -4,25 +4,32 @@ ms.date: "03/30/2017"
-  
-## Debugging in Code  
- Code breakpoints can be used in [!INCLUDE[netfx_current_short](../../../includes/netfx-current-short-md.md)] in the same way that they can be used in other imperative applications. Click the left margin of the code pane to create a code breakpoint, or press **F9** to place a breakpoint at the cursor location.  
-  
-## Attaching to a Workflow Process  
- Workflow debugging also supports using Visual Studio’s infrastructure to attach to a process. This enables the workflow author to debug a workflow running in a different host environment such as Internet Information Services (IIS) 7.0.  
-  
-## Remote Debugging  
- Windows Workflow Foundation (WF) remote debugging functions the same as remote debugging for other Visual Studio components. For information on using remote debugging, see [How to: Enable Remote Debugging](https://go.microsoft.com/fwlink/?LinkId=196257).  
-  
+
+[!INCLUDE[netfx_current_long](../../../includes/netfx-current-long-md.md)] offers several options for debugging running workflows from the development environment. Workflows can be debugged in the designer, in XAML, and in code.
+
+## Debugging in the Workflow Designer
+
+Breakpoints can be set on activities in the workflow designer by either highlighting the activity and pressing <kbd>F9</kbd> or by using the activity’s context menu. Execution of the workflow then breaks when the workflow host is run in debug mode. In the following screenshot, workflow execution is paused at a breakpoint. For more information, see [Debugging Workflows with the Workflow Designer](/visualstudio/workflow-designer/debugging-workflows-with-the-workflow-designer).
+
+## Debugging in XAML
+
+If a workflow has paused at a breakpoint in the designer, the workflow can also be debugged in XAML. To view the point of execution in XAML, select **XAML View** in the workflow designer when workflow execution is paused. Debugging can be switched back to the designer by re-opening the workflow in the designer from the solution explorer. For more information, see [How to: Debug XAML with the Workflow Designer](/visualstudio/workflow-designer/how-to-debug-xaml-with-the-workflow-designer).
+
+## Debugging in Code
+
+Code breakpoints can be used in [!INCLUDE[netfx_current_short](../../../includes/netfx-current-short-md.md)] in the same way that they can be used in other imperative applications. Click the left margin of the code pane to create a code breakpoint, or press <kbd>F9</kbd> to place a breakpoint at the cursor location.
```

Comment:
```suggestion
To set a breakpoint, click the left margin of the code pane to create a code breakpoint, or press <kbd>F9</kbd> to place a breakpoint at the cursor location.
```

Reply:
@rpetrusha for being late, just noticed this now.

I think the repetition of `set a breakpoint` and `create a breakpoint` is unintentional. so I'm modifying the suggestion to:

```suggestion
To set a breakpoint, click the left margin of the code pane, or press <kbd>F9</kbd> with the cursor at the line where you want to set it.
```